.devcontainer/build.sh

```diff
@@ -35,6 +35,8 @@ main() {
   if [ "$DISABLE_POSTBUILD" != "true" -a "$BUILD_RESULT" == 0 ]; then
     source "$BUILD_DIR/post_build.sh"
   fi
+  # assuming post_build.sh will never fail on a successful build
+  return $BUILD_RESULT
```

Comment:
I think this should `exit` instead of `return`. `exit` will stop running the script and return the correct value, but `return` will continue executing the script, so running any command after main would override the value. I would change this to avoid issues in the future.

Reply:
the script seems like it can be sourced as well. If someone would source it in another script an then call the `main` function it will terminate its shell as well. Which isn't behavior I'd expect from a function

Reply:
Ok. Then `exit` should be placed after the call to `main`.

Reply:
Explicit `exit` is very important to make the true intent of the code apparent and avoid bugs. Without any indication, it's very easy to break the workflow again, for example just by adding an echo after main for more verbose output. In fact, I think a comment should also be added unless the return value of main is explicitly used when exiting.

Reply:
Agreed, pushed an update with explicit exit code handling and a comment.